In [3]:
from sklearn.datasets import load_breast_cancer
import pandas as pd


data = load_breast_cancer()

X = pd.DataFrame(data.data, columns=data.feature_names)
y = data.target

X.head()

,mean radius,mean texture,mean perimeter,mean area,mean smoothness,mean compactness,mean concavity,mean concave points,mean symmetry,mean fractal dimension,...,worst radius,worst texture,worst perimeter,worst area,worst smoothness,worst compactness,worst concavity,worst concave points,worst symmetry,worst fractal dimension
0,17.99,10.38,122.80,1001.0,0.11840,0.27760,0.3001,0.14710,0.2419,0.07871,...,25.38,17.33,184.60,2019.0,0.1622,0.6656,0.7119,0.2654,0.4601,0.11890
1,20.57,17.77,132.90,1326.0,0.08474,0.07864,0.0869,0.07017,0.1812,0.05667,...,24.99,23.41,158.80,1956.0,0.1238,0.1866,0.2416,0.1860,0.2750,0.08902
2,19.69,21.25,130.00,1203.0,0.10960,0.15990,0.1974,0.12790,0.2069,0.05999,...,23.57,25.53,152.50,1709.0,0.1444,0.4245,0.4504,0.2430,0.3613,0.08758
3,11.42,20.38,77.58,386.1,0.14250,0.28390,0.2414,0.10520,0.2597,0.09744,...,14.91,26.50,98.87,567.7,0.2098,0.8663,0.6869,0.2575,0.6638,0.17300
4,20.29,14.34,135.10,1297.0,0.10030,0.13280,0.1980,0.10430,0.1809,0.05883,...,22.54,16.67,152.20,1575.0,0.1374,0.2050,0.4000,0.1625,0.2364,0.07678


In [4]:
X.isnull().sum()

mean radius                0
mean texture               0
mean perimeter             0
mean area                  0
mean smoothness            0
mean compactness           0
mean concavity             0
mean concave points        0
mean symmetry              0
mean fractal dimension     0
radius error               0
texture error              0
perimeter error            0
area error                 0
smoothness error           0
compactness error          0
concavity error            0
concave points error       0
symmetry error             0
fractal dimension error    0
worst radius               0
worst texture              0
worst perimeter            0
worst area                 0
worst smoothness           0
worst compactness          0
worst concavity            0
worst concave points       0
worst symmetry             0
worst fractal dimension    0
dtype: int64

In [6]:

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

The dataset was loaded using sklearn’s breast cancer dataset.
Missing values were checked using isnull().sum() and no missing values were found.
The dataset was split into training (80%) and testing (20%) sets.
Feature scaling was performed using StandardScaler to normalize the data.
Scaling is necessary because algorithms like SVM and KNN rely on distance calculations, and unscaled data can affect performance.

In [8]:
# Logistic Regression

from sklearn.linear_model import LogisticRegression

lr = LogisticRegression(max_iter=1000)
lr.fit(X_train, y_train)
lr_acc = lr.score(X_test, y_test)

print(lr_acc)

0.9736842105263158


Logistic Regression is a binary classification algorithm that uses a sigmoid function to predict probabilities.
It is suitable for this dataset because the target variable has two classes (benign and malignant).

In [10]:
# Decision Tree Classifier

from sklearn.tree import DecisionTreeClassifier

dt = DecisionTreeClassifier(random_state=42)
dt.fit(X_train, y_train)
dt_acc = dt.score(X_test, y_test)

print(dt_acc)

0.9473684210526315


Decision Tree splits the data into branches based on feature values.
It is suitable because it can model non-linear relationships and is easy to interpret.

In [12]:
# Random Forest Classifier
from sklearn.ensemble import RandomForestClassifier

rf = RandomForestClassifier(random_state=42)
rf.fit(X_train, y_train)
rf_acc = rf.score(X_test, y_test)

print(rf_acc)

0.9649122807017544


Random Forest is an ensemble method that combines multiple decision trees to improve accuracy.
It is suitable because it reduces overfitting and performs well on complex datasets.

In [13]:
# Support Vector Machine (SVM)

from sklearn.svm import SVC

svm = SVC()
svm.fit(X_train, y_train)
svm_acc = svm.score(X_test, y_test)

print(svm_acc)

0.9824561403508771


In [14]:
#k-Nearest Neighbors (KNN)

from sklearn.neighbors import KNeighborsClassifier

knn = KNeighborsClassifier()
knn.fit(X_train, y_train)
knn_acc = knn.score(X_test, y_test)

print(knn_acc)

0.9473684210526315


In [15]:
# comparison


import pandas as pd

results = pd.DataFrame({
    'Model': ['Logistic Regression', 'Decision Tree', 'Random Forest', 'SVM', 'KNN'],
    'Accuracy': [lr_acc, dt_acc, rf_acc, svm_acc, knn_acc]
})

print(results)

                 Model  Accuracy
0  Logistic Regression  0.973684
1        Decision Tree  0.947368
2        Random Forest  0.964912
3                  SVM  0.982456
4                  KNN  0.947368


The performance of all five classification algorithms was compared using accuracy.
Among all the models, Support Vector Machine (SVM) achieved the highest accuracy of approximately 98.25%, making it the best-performing model.

Decision Tree and KNN achieved the lowest accuracy of approximately 94.7%. Decision Tree tends to overfit the data, while KNN performance depends on distance calculations and data distribution.

Random Forest and Logistic Regression also performed well.
Thus, SVM is the most suitable model for this dataset, while Decision Tree and KNN performed the worst.